File: LC_JS-SOLUTIONS/2778. Sum of Squares of Special Elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var sumOfSquares = function(nums) {

    let n = nums.length
    res = 0
    for(let i=0;i<nums.length;i++ ){
        if(n%(i+1)==0){
            res+=nums[i]**2
        }
    }

    return res


    
};

File: LC-Python-Solution/2778. Sum of Squares of Special Elements.py
class Solution:
    def sumOfSquares(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0

        for i in range(0,len(nums)):
            if n%(i+1)==0:
                res+= nums[i]**2
        
        return res




# ANOTHER SOLUTION

def sumOfSquares(nums):
    n = len(nums)
    return sum(nums[i]**2 for i in range(0,len(nums)) if n%(i+1)==0)
    
    
    
    